File: app/src/main/java/com/frankegan/verdant/home/HomeContract.java
package com.frankegan.verdant.home;

import android.support.annotation.NonNull;

import com.frankegan.verdant.models.ImgurImage;

import java.util.List;

/**
 * Created by frankegan on 5/10/16.
 * <p>
 * This specifies the contract between the view and the presenter.
 */
public class HomeContract {

    interface View {

        void setProgressIndicator(boolean active);

        void showImages(List<ImgurImage> notes);

        void showImageDetailUi(ImgurImage image, android.view.View view);

        void showSubredditChooser();

        void clearImages();

        void showBottomSheet(boolean show);

        void refreshRecents();

        void setToolbarTitle(String title);
    }

    interface UserActionsListener {

        void loadMoreImages(int page);

        void openImageDetails(@NonNull ImgurImage requestedNote, android.view.View view);

        void changeSubreddit(String subName);

        String getSubreddit();
    }
}

File: app/src/main/java/com/frankegan/verdant/home/HomeContract.kt
package com.frankegan.verdant.home

import com.frankegan.verdant.models.ImgurImage

/**
 * Created by frankegan on 5/10/16.
 *
 *
 * This specifies the contract between the view and the presenter.
 */
class HomeContract {

    interface View {

        fun setProgressIndicator(active: Boolean)

        fun showImages(images: List<ImgurImage>)

        fun showImageDetailUi(image: ImgurImage, view: android.view.View)

        fun showSubredditChooser()

        fun clearImages()

        fun showBottomSheet(show: Boolean)

        fun refreshRecents()

        fun setToolbarTitle(title: String)
    }

    internal interface UserActionsListener {

        val subreddit: String

        fun loadMoreImages(page: Int)

        fun changeSubreddit(subName: String)

    }
}
